libdevcore/Common.h

```diff
@@ -91,6 +91,24 @@ inline u256 s2u(s256 _u)
 		return u256(c_end + _u);
 }
 
+/// @returns the highest bit set in @a. If _v is zero, returns -1.
+inline int highestBitSet(u256 const& _v)
+{
+	int bit = 255;
+	while (bit >= 0 && (((_v >> bit) & 1) == 0))
+		bit--;
+	return bit;
+}
+
+/// @returns the lowest bit set in @a. If _v is zero, returns 256
+inline int lowestBitSet(u256 const& _v)
```

Comment:
Possible alternative:
```
if (n == 0)
  return 256;
return log2(n & ~(n-1));
```
`n & ~(n-1)` clears all the bits except the last, so we could potentially implement the `log2` version for powers of 2 in `logN` time (which is not very elegant btw)